controllers/openstackcontrolplane_controller.go

```diff
@@ -478,13 +463,135 @@ func (r *OpenStackControlPlaneReconciler) createOrGetTripleoPasswords(
 
 			return err
 		}
+	} else {
+		tripleoPasswordsByte := secret.Data["tripleo-overcloud-passwords.yaml"]
+
+		tripleoPasswordsRaw := make(map[string]interface{})
+		if err := yaml.Unmarshal(tripleoPasswordsByte, &tripleoPasswordsRaw); err != nil {
+			cond.Message = fmt.Sprintf("Error extract TripleoPasswords from Secret %s", controlplane.TripleoPasswordSecret)
+			cond.Reason = shared.ControlPlaneReasonTripleoPasswordsSecretError
+			cond.Type = shared.CommonCondTypeError
+			err = common.WrapErrorForObject(cond.Message, instance, err)
+
+			return err
+		}
+
+		// recursice convert nested map[interface{}]interface{} password data
+		// into map[string]interface{} and merge into a new tripleoPasswords map
+		tripleoPasswords, err := common.RecursiveMergeMaps(
+			map[string]interface{}{},
+			tripleoPasswordsRaw,
+			common.PasswordMaxDepth,
+		)
```

Comment:
Just want to make sure I understand the comment.  So the `tripleoPasswordsRaw` variable is `map[string]interface{}` but the `interface{}` values themselves are `map[interface{}]interface{}`, so it's really `map[string]map[interface{}]interface{}?  That's my guess, at least, because it seems like you already have `map[string]interface{}`.

Reply:
yes correct, nested maps are of type map[interface{}]interface{}, but how many nested maps there are is unknown.

Reply:
I would have thought map[string]interface{} would be enough here. Didn't realize we were nesting that deep

Reply:
its the nested jsons which gets marshalled into nested map[interface{}]interface{}.